CRM/Extension/Browser.php

```diff
@@ -157,21 +157,31 @@ public function getExtensions() {
 
   /**
    * Get a description of a particular extension.
+   * This function is an optimised form of getExtensions() - it returns the
+   * extension's details as soon as it's found without caching anything.
    *
    * @param string $key
    *   Fully-qualified extension name.
    *
    * @return CRM_Extension_Info|NULL
    */
   public function getExtension($key) {
-    // TODO optimize performance -- we don't need to fetch/cache the entire repo
-    $exts = $this->getExtensions();
-    if (array_key_exists($key, $exts)) {
-      return $exts[$key];
+    if (!$this->isEnabled() || count($this->checkRequirements())) {
+      return NULL;
     }
-    else {
+
+    $remote = $this->_discoverRemote();
```

Comment:
I may be failing to see the logic here, but I thought the purpose of this PR was to avoid fetching the entire list of extensions? @totten, since you left the comment about optimizing maybe you could weigh in here.